beetsplug/lastgenre/__init__.py

```diff
@@ -348,26 +387,40 @@ def _get_genre(self, obj):
                     if not item_genre:
                         item_genre = self.fetch_artist_genre(item)
                     if item_genre:
-                        item_genres.append(item_genre)
+                        item_genres += item_genre
                 if item_genres:
-                    result, _ = plurality(item_genres)
+                    most_popular, rank = plurality(item_genres)
+                    new_genres = [most_popular]
+                    label = (
+                        "most popular track, whitelist"
+                        if self.whitelist
+                        else "most popular track, any"
+                    )
+                    self._log.debug(
+                        'Most popular track genre "{}" ({}) for VA album.',
+                        most_popular,
+                        rank,
+                    )
 
-            if result:
-                return result, "artist"
+        # Return with a combined or freshly fetched genre list.
+        if new_genres:
+            if keep_genres:
```

Comment:
My final comment: I noticed that these lines share the same type of logic

```python
            label = "track, whitelist" if self.whitelist else "track, any"
            label = "album, whitelist" if self.whitelist else "album, any"
            label = "artist, whitelist" if self.whitelist else "artist, any"
            label = "album artist, whitelist" if self.whitelist else "album artist, any"
            label = "most popular track, whitelist" if self.whitelist else "most popular track, any"
```

Could you revert those lines to what we had here before, i.e. `label = "track"` and perform the whitelist check just once, somewhere around here?

```suggestion
        if new_genres:
            suffix = "whitelist" if self.whitelist else "any"
            label += f", {suffix}"

            if keep_genres:
```

Reply:
That is a great suggestion. To be honest during adding these lines I had thoughts of "maybe polish this later" but probably would have forgotten about it! Many thanks! That helps!